maven_iuvs/echelle.py

```diff
@@ -334,11 +334,163 @@ def downselect_data(index, light_dark=None, orbit=None, date=None, segment=None,
 
 # Relating to dark vs. light observations -----------------------------
 
-def update_master_lightdark_key(key_filename, ech_l1a_idx, dark_idx, 
+def update_filenames_in_light_dark_key(keyfile, ech_l1a_idx, dark_idx, 
+                                       verbose=True):
+    """
+    Searches the light/dark key csv (keyfile) for any filenames that have since 
+    had a revision update. In that sense, the "copy of record" is the copy on 
+    the user's local hard drive. 
+
+    Parameters
+    ----------
+    keyfile : string
+              .csv file containing columns 'Light Folder', 'Light', 'Dark  
+              Folder', 'Dark', 'DTobj, 'Segment'.
+    ech_l1a_idx : list of dictionaries
+                  Metadata of local files, used to determine newest filename.
+    dark_idx : list of dictionaries
+               Same as above but the dark files only.
+    verbose : bool
+              If True will print result messages.
+
+    Returns
+    ----------
+    MASTER_KEY : pandas DataFrame
+                 New master key. 10s of thouasnds of lines
+    changes : pandas DataFrame
+              Lists old files and what they were replaced with
+    problems : pandas DataFrame
+               Files that caused a problem
+    """
+    MASTER_KEY = pd.read_csv(keyfile)
+
+    changes = {"row": [], "oldname": [], "newname": []}
+    problems = {"row": [], "oldname": [], "newname": []}
+
+    # Precompile regex
+    versioned_pat = re.compile(r"^(?P<base>mvn.+?)_(r|s)(?P<rev>\d+)(?P<ext>\.fits\.gz)$")
```

Comment:
you're actually gonna make everyone who maintains this repo have to learn regular expressions, aren't you??? :sob: :cries due to laziness:

Reply:
https://xkcd.com/208/